src/file_dialog.rs

```diff
@@ -45,7 +45,7 @@ impl FileDialog {
     ///   * Linux
     ///
     /// On platforms that don't support filter names, all filters will be merged into one filter
-    pub fn add_filter(mut self, name: &str, extensions: &[&str]) -> Self {
+    pub fn add_filter(mut self, name: impl Into<String>, extensions: &[impl ToString]) -> Self {
```

Comment:
Could you elaborate on the choice of ToString for extensions?
The idea is that those could be stored somewhere by the consumer across dialog creations?